classes/amity.py

```diff
@@ -1,45 +1,515 @@
+            output += (" ID   " + "  Person name" + "\n")
+            for name in unallocated_lspace:
+                output += ("{} -> {} \n".format(name.person_id,
+                           name.person_name))
+        else:
+            output += "\nNo one in the living_space waiting list"
+        if file_name is None:
+            print(output)
+            return "Data printed successfully\n"
+        else:
+            save_to = open(file_name + ".txt", "w")
+            save_to.write(output)
+            save_to.close()
+            return "Data saved in {} successfully".format(file_name)
 
-    def print_room(self, room_name):
-        """A method that prints room occupants in a room"""
-        pass
+    def allocate_person_office(self, person_id):
+        """A method that allocates a person in the waiting list to an office"""
+        person_id = person_id.upper()
+        found = False
+        person_obj = None
+        for person in self.waiting_list["office"]:
+            if person.person_id == person_id:
+                found = True
+                person_obj = person
+                self.allocate_office(person)
+                break
+        if not found:
+            return "The person is not in the office waiting list"
+        if person_obj.office is None:
+            return "No available offices"
+        else:
+            self.waiting_list["office"].remove(person_obj)
+            return "{} has been allocated to office {}".format(
+             person_obj.person_name, person_obj.office.room_name)
+
+    def allocate_person_livingspace(self, person_id):
+        """A method that allocates a person in the waiting list a livingspace"""
+        person_id = person_id.upper()
+        found = False
+        person_obj = None
+        for person in self.waiting_list["livingspace"]:
+            if person.person_id == person_id:
+                found = True
+                person_obj = person
+                self.allocate_living_space(person)
+                break
+        if not found:
+            return "The person is not in the livingspace waiting list"
+        if person_obj.living_space is None:
+            return "No available livingspaces"
+        else:
+            self.waiting_list["livingspace"].remove(person_obj)
+            return "{} has been allocated to livingspace {}".format(
+             person_obj.person_name, person_obj.living_space.room_name)
+
+    def loads_people(self, file_name):
+        """A method that adds people from a text file"""
+        try:
+            read_file = open(file_name + '.txt', 'r')
+            people = read_file.readlines()
+            if len(people) == 0:
+                return 'Empty file. No one has been added.'
+            for person in people:
+                splitwords = person.split()
+                if len(splitwords) <= 3:
+                    wants_accommodation = 'n'
+                else:
+                    wants_accommodation = splitwords[3]
```

Comment:
You might want to catch an IndexError here in case the text file has invalid format in terms of number of arguments.